aea/decision_maker/base.py

```diff
@@ -319,36 +323,49 @@ def _handle_tx_message(self, tx_message: TransactionMessage) -> None:
         :param tx_message: the transaction message
         :return: None
         """
+        # get variables
         crypto_identifier = CURRENCY_TO_ID_MAP.get(cast(str, tx_message.get("currency_pbk")))
-        amount = cast(float, tx_message.get("amount"))
         crypto_object = self._wallet.crypto_objects.get(crypto_identifier)
-        counterparty_address = crypto_object.generate_counterparty_address(cast(str, tx_message.get("counterparty")))
-
-        # check if the transaction is acceptable
-        if self._is_acceptable_tx(crypto_object, amount):
-            tx_digest = self._settle_tx(crypto_object, counterparty_address, amount, cast(float, tx_message.get("sender_tx_fee")))
+        amount = cast(int, tx_message.get("amount"))
+        counterparty_tx_fee = cast(int, tx_message.get("counterparty_tx_fee"))
+        sender_tx_fee = cast(int, tx_message.get("sender_tx_fee"))
+        counterparty_address = cast(str, tx_message.get("counterparty"))
+
+        # adjust payment amount to reflect transaction fee split
+        amount -= counterparty_tx_fee
+        tx_fee = counterparty_tx_fee + sender_tx_fee
+        payable = amount + tx_fee
+
+        # check if the transaction is acceptable and process it accordingly
+        if self._is_acceptable_tx(crypto_object, payable):
+            tx_digest = self._settle_tx(crypto_object, counterparty_address, amount, tx_fee)
```

Comment:
This is currently blocking. We can keep it that way for now I think. We should address this in the context of the actor model/async discussions.